pkg/trace/writer/trace.go

```diff
@@ -160,6 +195,22 @@ func (w *TraceWriter) addSpans(pkg *SampledSpans) {
 	w.bufferedSize += size
 }
 
+func (w *TraceWriter) drainAndFlush() {
+outer:
+	for {
+		select {
+		case pkg := <-w.In:
+			w.addSpans(pkg)
+		default:
+			break outer
+		}
+	}
+	w.flush()
+	// Wait for encoding/compression to complete on each payload,
+	// and submission to senders
+	w.wg.Wait()
```

Comment:
I think you forgot to remove this line from the `Stop` method, now that we're doing it here, it shouldn't be needed there. Right?